datafusion/proto/src/common.rs

```diff
@@ -17,26 +17,17 @@
 
 use datafusion_common::{DataFusionError, Result};
 
-pub fn csv_delimiter_to_string(b: u8) -> Result<String> {
-    let b = &[b];
-    let b = std::str::from_utf8(b)
-        .map_err(|_| DataFusionError::Internal("Invalid CSV delimiter".to_owned()))?;
-    Ok(b.to_owned())
-}
-
-pub fn str_to_byte(s: &String) -> Result<u8> {
+pub fn str_to_byte(s: &String, flag: &str) -> Result<u8> {
     if s.len() != 1 {
-        return Err(DataFusionError::Internal(
-            "Invalid CSV delimiter".to_owned(),
-        ));
+        return Err(DataFusionError::Internal(format!("Invalid CSV {flag}")));
```

Comment:
I think more specific error messages are more helpful:

```suggestion
        return Err(DataFusionError::Internal(format!("Invalid CSV {flag}: expected single character, got {s}")));
```